src/System.Private.ServiceModel/tests/Common/Infrastructure/Platform.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace Infrastructure.Common
+{
+    public static class Platform
+    {
+        private static PlatformID _currentPlatformID = 0;
+        public static PlatformID Current
+        {
+            get
+            {
+                if (_currentPlatformID == 0)
+                {
+                    _currentPlatformID = DetectCurrentPlatform();
+                }
+
+                return _currentPlatformID;
+            }
+        }
+
+        public static bool MatchesCurrentPlatform(PlatformID id)
+        {
+            return ((Current & id) != 0);
+        }
+
+        public static string CurrentName
+        {
+            get
+            {
+                return GetPlatformName(Current);
+            }
+        }
+
+        public static string GetPlatformName(PlatformID id)
+        {
+            return Enum.Format(typeof(PlatformID), id, "G");
+        }
```

Comment:
Could this be an extension method for PlatformID? That way you could get rid of the CurrentName property and instead do the following.
```csharp
    Platform.Current.GetName();
```